src/dma.rs

```diff
@@ -161,32 +185,32 @@ macro_rules! dma {
                             }
                         }
 
-                        pub(crate) fn isr(&self) -> dma1::isr::R {
+                        fn isr(&self) -> dma1::isr::R {
                             // NOTE(unsafe) atomic read with no side effects
                             unsafe { (*$DMAX::ptr()).isr.read() }
                         }
 
-                        pub(crate) fn ifcr(&self) -> &dma1::IFCR {
+                        fn ifcr(&self) -> &dma1::IFCR {
                             unsafe { &(*$DMAX::ptr()).ifcr }
                         }
 
-                        pub(crate) fn ccr(&mut self) -> &dma1::$CCRX {
+                        fn ccr(&mut self) -> &dma1::$CCRX {
```

Comment:
I'd prefer `fn ccr(&mut self) -> &Self::CCRX`, same for the other implementations.